homeassistant/components/withings/config_flow.py

```diff
@@ -0,0 +1,242 @@
+"""Config flow for Withings."""
+from collections import OrderedDict
+import logging
+from typing import Optional
+
+import nokia
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.util import slugify
+
+from . import const
+
+DATA_FLOW_IMPL = 'withings_flow_implementation'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def auth_callback_path(profile: str) -> str:
+    """Create an auth callback path."""
+    return '{}/{}'.format(const.AUTH_CALLBACK_PATH, slugify(profile))
+
+
+def auth_callback_name(profile: str) -> str:
+    """Create an auth callback name."""
+    return '{}:{}'.format(const.AUTH_CALLBACK_NAME, slugify(profile))
+
+
+@callback
+def register_flow_implementation(
+        hass,
+        client_id,
+        client_secret,
+        base_url,
+        profile
+):
+    """Register a flow implementation.
+
+    domain: Domain of the component responsible for the implementation.
+    name: Name of the component.
+    client_id: Client id.
+    client_secret: Client secret.
+    """
+    if DATA_FLOW_IMPL not in hass.data:
+        hass.data[DATA_FLOW_IMPL] = OrderedDict()
+
+    hass.data[DATA_FLOW_IMPL][profile] = {
+        const.CLIENT_ID: client_id,
+        const.CLIENT_SECRET: client_secret,
+        const.BASE_URL: base_url,
+        const.PROFILE: profile,
+    }
+
+
+@config_entries.HANDLERS.register(const.DOMAIN)
+class WithingsFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self.flow_profile = None
+
+    def async_profile_config_entry(
+            self,
+            profile: str
+    ) -> Optional[ConfigEntry]:
+        """Get a profile config entry."""
+        entries = self.hass.config_entries.async_entries(const.DOMAIN)
+        for entry in entries:
+            if entry.title == profile:
```

Comment:
In the future users might be able to change the title. Compare to data stored in `entry.data` instead.